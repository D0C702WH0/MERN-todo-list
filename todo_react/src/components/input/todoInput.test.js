import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TodoInput from './todoInput';

Enzyme.configure({ adapter: new Adapter() });

let input;
let todoAdd;
beforeEach(() => {
  todoAdd = jest.fn(),
    input = shallow(<TodoInput todoAdd={todoAdd} />);
});

describe('Input', () => {

  test('it renders correctly', () => {
    expect(input).toMatchSnapshot();
  });

  it('Should change input state with handleChange()', () => {

    const inputTodo = input.find('#todoInput');
    const mockEvent = {
      target: {
        name: 'input',
        value: "test"
      },
    };
    inputTodo.simulate('change', mockEvent);
    expect(input.state().input).toEqual('test');
  });


  it('Should call todoAdd() on submit', () => {
    const form = input.find('form');
    const handleSubmit = jest.fn();
    form.simulate('submit', {
      preventDefault: () => {
      }
    });
    const todoAdd = input.instance().props.todoAdd;

    expect(todoAdd).toHaveBeenCalled();
  });

  it('Should reset input state with handleSubmit()', () => {
    const btn = input.find('#submit');
    btn.simulate('click');
    expect(input.state().input).toEqual('');
  });
})
