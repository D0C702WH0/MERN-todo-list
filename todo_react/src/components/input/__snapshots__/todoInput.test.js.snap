// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input it renders correctly 1`] = `ShallowWrapper {}`;

exports[`Input renders correctly it renders correctly 1`] = `ShallowWrapper {}`;
